### PR TITLE
PMM-6623 Listing all metrics in tests

### PR DESCRIPTION
**Pull request by percona-csalguero**

### CODE REVIEW

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Looks good to me, but do we have list of metrics for compatible mode?
Can we write test for compatible mode as well?
Please fix tests.

File: exporter/diagnostic_data_collector_test.go [hunk 1]
```diff
@@ -77,45 +77,48 @@ mongodb_oplog_stats_wt_transaction_update_conflicts 0` + "\n")
 }
 
 func TestAllDiagnosticDataCollectorMetrics(t *testing.T) {
+	if inGithubActions() {
+		t.Skip("Test not reliable in Gihub Actions")
```
Comment (BupycHuk) [inline exporter/diagnostic_data_collector_test.go:81]: Maybe it's not a good idea to rewrite tests if they are not reliable in Github actions.

File: exporter/replset_status_collector_test.go [hunk 2]
```diff
@@ -72,21 +72,23 @@ mongodb_optimes_durableOpTime_t 1` + "\n")
 	assert.NoError(t, err)
 }
 
-func TestReplsetStatusCollectorNoSharding(t *testing.T) {
+func TestAllReplsetStatusCollectorMetrics(t *testing.T) {
+	if inGithubActions() {
+		t.Skip("Test not reliable in Gihub Actions")
```
Comment (BupycHuk) [inline exporter/replset_status_collector_test.go:77]: The same here

File: exporter/serverstatus_collector_test.go [hunk 3]
```diff
@@ -63,3 +63,24 @@ mongodb_metrics_commands_connPoolSync_failed 0` + "\n")
 	err := testutil.CollectAndCompare(c, expected, filter...)
 	assert.NoError(t, err)
 }
+
+func TestAllServerStatusDataCollector(t *testing.T) {
+	if inGithubActions() {
+		t.Skip("Test not reliable in Gihub Actions")
```
Comment (BupycHuk) [inline exporter/serverstatus_collector_test.go:69]: and here

File: exporter/diagnostic_data_collector_test.go (hunk 1 above)
Comment (BupycHuk) [reply exporter/diagnostic_data_collector_test.go:81]: Can we make some queries in mongodb server and try to get full list of metrics?

**Review by BupycHuk — CHANGES_REQUESTED** [review]
I don't think that skipping tests in github actions is a good idea.

File: exporter/diagnostic_data_collector_test.go (hunk 1 above)
Comment (percona-csalguero) [reply exporter/diagnostic_data_collector_test.go:81]: I already tried and it is different in github actions and locally

Comment (percona-csalguero) [reply exporter/diagnostic_data_collector_test.go]: Updated. Created a base list of metrics, common to all versions.

File: exporter/exporter.go [hunk 4]
```diff
@@ -62,7 +63,7 @@ var (
 // New connects to the database and returns a new Exporter instance.
 func New(opts *Opts) (*Exporter, error) {
 	if opts == nil {
-		opts = new(Opts)
+		opts = &Opts{}
```
Comment (BupycHuk) [inline exporter/exporter.go:66]: Any reason to use this style?

File: exporter/replset_status_collector_test.go [hunk 5]
```diff
@@ -72,21 +72,23 @@ mongodb_optimes_durableOpTime_t 1` + "\n")
...
-		ctx:          ctx,
-		client:       client,
-		topologyInfo: ti,
-	}
-
-	expected := strings.NewReader(``)
-	err := testutil.CollectAndCompare(c, expected)
-	assert.NoError(t, err)
-}
+// func TestAllReplsetStatusCollectorMetrics(t *testing.T) {
```
Comment (BupycHuk) [inline exporter/replset_status_collector_test.go:75]: Why it's commented?

File: exporter/serverstatus_collector_test.go [hunk 6]
```diff
@@ -63,3 +64,26 @@ mongodb_metrics_commands_connPoolSync_failed 0` + "\n")
 	err := testutil.CollectAndCompare(c, expected, filter...)
 	assert.NoError(t, err)
 }
+
+func TestAllServerStatusDataCollector(t *testing.T) {
+	if inGithubActions() {
```
Comment (BupycHuk) [inline exporter/serverstatus_collector_test.go:69]: Couldn't we make this test stable in github actions?

File: exporter/exporter.go (hunk 4 above)
Comment (percona-csalguero) [reply exporter/exporter.go:66]: go-consistent

File: exporter/serverstatus_collector_test.go (hunk 6 above)
Comment (percona-csalguero) [reply exporter/serverstatus_collector_test.go:69]: No

File: exporter/replset_status_collector_test.go (hunk 5 above)
Comment (percona-csalguero) [reply exporter/replset_status_collector_test.go]: Re-enabled

**Comment by denisok** [comment]
Hello PR owner,

we are cleaning up the mess with the branches. New [mongodb_exporter](https://www.percona.com/blog/2020/09/22/new-mongodb-exporter-released-with-percona-monitoring-and-management-2-10-0/) implementation would be default `main` branch.
Old `master` branch is moved to the `release-0.1x` branch.

Please resubmit and resolve any conflicts against the `main` and we would start to do the better job to get them merged to the `main` and produce regular releases.

We also moved to the Github Actions from Travis CI, so that would check your code going forward.

Thanks,
Denys from Percona

**Pull request closed without merging**